ros2bag/ros2bag/verb/record.py

```diff
@@ -167,20 +184,41 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             help='Choose the compression format/algorithm. '
                  'Has no effect if no compression mode is chosen. Default: %(default)s.')
 
+    def _check_necessary_argument(self, args):
+        # One options out of --all, --all-topics, --all-services, --services, topics or --regex
+        # must be used
+        if not (args.all or args.all_topics or args.all_services or
+           args.services or (args.topics and len(args.topics) > 0) or args.regex):
+            return False
+        return True
+
     def main(self, *, args):  # noqa: D102
-        # both all and topics cannot be true
-        if (args.all and (args.topics or args.regex)) or (args.topics and args.regex):
-            return print_error('Must specify only one option out of topics, --regex or --all')
-        # one out of "all", "topics" and "regex" must be true
-        if not(args.all or (args.topics and len(args.topics) > 0) or (args.regex)):
-            return print_error('Invalid choice: Must specify topic(s), --regex or --all')
 
-        if args.topics and args.exclude:
-            return print_error('--exclude argument cannot be used when specifying a list '
-                               'of topics explicitly')
+        if not self._check_necessary_argument(args):
+            return print_error('Must specify only one option out of --all, --all-topics, '
+                               '--all-services, --services, topics and --regex')
```

Comment:
Comments from MichaelOrlov

```suggestion
        if not self._check_necessary_argument(args):
            return print_error('Need to specify one option out of --all, --all-topics, '
                               '--all-services, --services, topics and --regex')
```

Reply:
Updated